resources/nspawn.go

```diff
@@ -0,0 +1,351 @@
+	// var activeSet = false
+	// var previous bool
+
+	for {
+		obj.SetState(ResStateWatching)
+		select {
+		// If this resource has been idle for long enough, set converged to allow
+		// the main loop to act on a converged status.
+		case <-cuid.ConvergedTimer():
+			cuid.SetConverged(true) // converged!
+			continue
+
+		case event := <-obj.Events():
+			cuid.SetConverged(false)
+			// we avoid sending events on unpause
+			if exit, send = obj.ReadEvent(&event); exit {
+				return nil // exit
+			}
+
+		case <-Startup(startup):
+			cuid.SetConverged(false)
+			send = true
+
+		// Process org.freedesktop.machine1 events for this resource's name.
+		case event := <-buschan:
+			if event.Body[0] == obj.GetName() {
+				if event.Name == machineNew {
+					log.Printf("Machine Started: %#v", event.Body[0])
+				}
+				if event.Name == machineRemoved {
+					log.Printf("Machine Stopped: %#v", event.Body[0])
+				}
+			}
+			send = true
+		}
+
+		// do all our event sending all together to avoid duplicate msgs
+		if send {
+			startup = true // startup finished
+			send = false
+			// only do this on certain types of events
+			//obj.isStateOK = false // something made state dirty
+			if exit, err := obj.DoSend(processChan, ""); exit || err != nil {
+				return err // we exit or bubble up a NACK...
+			}
+		}
+	}
+}
+
+// CheckApply is run to check the state and, if apply is true, to apply the necessary changes to reach the desired
+// state. This is run before Watch and again if watch finds a change occurring to the state.
+func (obj *NspawnRes) CheckApply(apply bool) (checkok bool, err error) {
+	log.Printf("%v[%v]: CheckApply(%t)", obj.Kind(), obj.GetName(), apply)
+
+	if obj.isStateOK { // cache the state
+		return true, nil
+	}
+
+	// This resource depends on systemd. Ensure that it's running.
+	// TODO: If systemd is not running, should we start it?
+	// TODO: We do this in two different places. Should we do it in Init?
+	if !systemdUtil.IsRunningSystemd() {
+		return false, errors.New("Systemd is not running.")
+	}
+
+	// Connect to org.freedesktop.machine1.Manager
+	conn, err := machined.New()
+
+	if err != nil {
+		return false, fmt.Errorf("Failed to connect to dbus: %s", err)
+	}
+
+	// Compare the current state with the desired state and perform the
+	// appropriate action
+	properties, err := conn.GetProperties(obj.GetName())
+	if err != nil {
+		if err.(dbus.Error).Name != "org.freedesktop.machine1.NoSuchMachine" {
+			log.Printf("GetProperties error: %s", err.Error())
+			return false, err
+		}
+		// If we could not successfully get the properties because there's no such machine, the machine
+		// is stopped.
+		if obj.State == stopped {
+			properties = make(map[string]interface{}, 1)
+			properties["State"] = stopped
+		}
+		// error if we need the image, ignore if we don't
+		_, err = conn.GetImage(obj.GetName())
+		if err != nil && obj.State != stopped {
+			return false, fmt.Errorf("No machine nor image named '%s'", obj.GetName())
```

Comment:
But I don't care what the err from GetImage was. This is a compound error representing the results of both GetMachine and GetImage so it is purposefully not returning the err from either.